### PR TITLE
Add DSS Component to the Micro Integrator

### PR DESCRIPTION
**Pull request by sajinieKavindya**
Moved from carbon-kernel

- org.wso2.carbon.ndatasource.core to org.wso2.micro.integrator.ndatasource.core
- org.wso2.carbon.ndatasource.common to org.wso2.micro.integrator.ndatasource.common
- org.wso2.carbon.ndatasource.capp.deployer to org.wso2.micro.integrator.ndatasource.capp.deployer
- org.wso2.carbon.ndatasource.rdbms to org.wso2.micro.integrator.ndatasource.rdbms

from carbon-data

- org.wso2.carbon.dataservices.core to org.wso2.micro.integrator.dataservices.core
- org.wso2.carbon.dataservices.common to org.wso2.micro.integrator.dataservices.common
- org.wso2.carbon.dataservices.odata to org.wso2.micro.integrator.dataservices.odata
- org.wso2.carbon.dataservices.odata.endpoint to org.wso2.micro.integrator.dataservices.odata.endpoint
- org.wso2.carbon.dataservices.capp.deployer to org.wso2.micro.integrator.dataservices.capp.deployer
- org.wso2.carbon.dataservices.sql.driver to org.wso2.micro.integrator.dataservices.sql.driver

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -671,6 +671,527 @@
...
+<!--            </dependency>-->
+            <dependency>
+                <groupId>org.eclipse.equinox</groupId>
+                <artifactId>javax.servlet</artifactId>
+                <version>${javax.servlet.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>javax.servlet</groupId>
+                <artifactId>servlet-api</artifactId>
+                <version>2.5</version>
```
Comment (Nirothipan) [inline pom.xml]: let's define a property for this

File: pom.xml [hunk 2]
```diff
@@ -671,6 +671,527 @@
...
+<!--            </dependency>-->
+<!--            <dependency>-->
+<!--                <groupId>io.netty</groupId>-->
+<!--                <artifactId>netty-codec</artifactId>-->
+<!--                <version>${netty.version}</version>-->
+<!--            </dependency>-->
+            <dependency>
+                <groupId>org.wso2.orbit.org.antlr</groupId>
+                <artifactId>antlr4-runtime</artifactId>
+                <version>4.5.1.wso2v1</version>
```
Comment (Nirothipan) [inline pom.xml]: let's define a property for this or use the existing one if there are any

File: pom.xml [hunk 3]
```diff
@@ -671,6 +671,527 @@
...
+                <artifactId>org.wso2.carbon.transaction.manager.feature</artifactId>
+                <type>zip</type>
+                <version>${carbon-commons.version}</version>
+            </dependency>
+<!--            <dependency>-->
+<!--                <groupId>org.apache.ws.commons.axiom.wso2</groupId>-->
+<!--                <artifactId>axiom</artifactId>-->
+<!--                <version>${axiom.wso2.version}</version>-->
+<!--            </dependency>-->
+<!--            <dependency>-->
```
Comment (Nirothipan) [inline pom.xml]: let's remove this commented lines

File: components/data/data-services/org.wso2.micro.integrator.dataservices.common/pom.xml [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    <modelVersion>4.0.0</modelVersion>
+    <artifactId>org.wso2.micro.integrator.dataservices.common</artifactId>
+    <packaging>bundle</packaging>
+    <name>WSO2 Carbon - Data Services Common</name>
+    <url>http://wso2.org</url>
+    <description>WSO2 Data Services Common Bundle</description>
+
+    <build>
+        <plugins>
+            <!--<plugin>-->
```
Comment (chanikag) [inline components/data/data-services/org.wso2.micro.integrator.dataservices.common/pom.xml]: Let's remove commented lines

File: components/data/data-services/org.wso2.micro.integrator.dataservices.common/src/main/java/org/wso2/micro/integrator/dataservices/common/RDBMSUtils.java [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+            } else if (JDBCDriverPrefixes.H2.equals(m.group())) {
+                return DBConstants.RDBMSEngines.H2;
+            } else if (JDBCDriverPrefixes.INFORMIX.equals(m.group())) {
+                return DBConstants.RDBMSEngines.INFORMIX_SQLI;
+            }
+        }
+        return DBConstants.RDBMSEngines.GENERIC;
+
+
+//		if (JDBCDriverPrefixes.MYSQL.indexOf(jdbcUrl) > -1) {
```
Comment (chanikag) [inline components/data/data-services/org.wso2.micro.integrator.dataservices.common/src/main/java/org/wso2/micro/integrator/dataservices/common/RDBMSUtils.java]: Let's remove commented lines

File: components/org.wso2.micro.integrator.initializer/pom.xml [hunk 6]
```diff
@@ -105,6 +105,18 @@
             <groupId>org.wso2.ei</groupId>
             <artifactId>org.wso2.micro.integrator.inbound.endpoint</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.wso2.ei</groupId>
+            <artifactId>org.wso2.micro.integrator.ndatasource.capp.deployer</artifactId>
+            <version>1.1.0-SNAPSHOT</version>
```
Comment (chanikag) [inline components/org.wso2.micro.integrator.initializer/pom.xml]: let's move version to parent pom

File: components/org.wso2.micro.integrator.ndatasource.capp.deployer/pom.xml [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+            <artifactId>org.wso2.micro.integrator.ndatasource.core</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.ws.commons.axiom.wso2</groupId>
+            <artifactId>axiom</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.wso2.ei</groupId>
+            <artifactId>org.wso2.micro.integrator.core</artifactId>
+            <version>1.1.0-SNAPSHOT</version>
```
Comment (chanikag) [inline components/org.wso2.micro.integrator.ndatasource.capp.deployer/pom.xml]: Let's move version to parent pom

File: components/org.wso2.micro.integrator.ndatasource.rdbms/pom.xml [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>4.13-beta-3</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.tomcat.wso2</groupId>
+            <artifactId>jdbc-pool</artifactId>
+            <version>7.0.34.wso2v2</version>
```
Comment (chanikag) [inline components/org.wso2.micro.integrator.ndatasource.rdbms/pom.xml]: Let's move version to parent pom

File: pom.xml (hunk 1 above)
Comment (sajinieKavindya) [reply pom.xml]: Done

File: pom.xml (hunk 2 above)
Comment (sajinieKavindya) [reply pom.xml]: Done

File: pom.xml (hunk 3 above)
Comment (sajinieKavindya) [reply pom.xml]: Done

File: components/data/data-services/org.wso2.micro.integrator.dataservices.common/pom.xml (hunk 4 above)
Comment (sajinieKavindya) [reply components/data/data-services/org.wso2.micro.integrator.dataservices.common/pom.xml]: Done

File: components/data/data-services/org.wso2.micro.integrator.dataservices.common/src/main/java/org/wso2/micro/integrator/dataservices/common/RDBMSUtils.java (hunk 5 above)
Comment (sajinieKavindya) [reply components/data/data-services/org.wso2.micro.integrator.dataservices.common/src/main/java/org/wso2/micro/integrator/dataservices/common/RDBMSUtils.java]: Done

File: components/org.wso2.micro.integrator.initializer/pom.xml (hunk 6 above)
Comment (sajinieKavindya) [reply components/org.wso2.micro.integrator.initializer/pom.xml]: Done

File: components/org.wso2.micro.integrator.ndatasource.capp.deployer/pom.xml (hunk 7 above)
Comment (sajinieKavindya) [reply components/org.wso2.micro.integrator.ndatasource.capp.deployer/pom.xml]: Done

File: components/org.wso2.micro.integrator.ndatasource.rdbms/pom.xml (hunk 8 above)
Comment (sajinieKavindya) [reply components/org.wso2.micro.integrator.ndatasource.rdbms/pom.xml]: Done

**Comment by Nirothipan** [comment]
Moved to https://github.com/wso2/micro-integrator/pull/810

**Pull request closed without merging**